### PR TITLE
build: resolve all floating promise warnings

### PR DESCRIPTION
**Pull request by samsiegart**
refs: [#6000](https://github.com/Agoric/agoric-sdk/issues/6000)

## Description

Resolve all violations of `@typescript-eslint/no-floating-promises` in all packages (except for `SwingSet` and `swingset-runner`, see discussion below).

Change rule from `warn` to `error` to ensure more instances don't come up after this effort.

### Security Considerations

N/A

### Scaling Considerations

N/A

### Documentation Considerations

N/A

### Testing Considerations

Ran `NODE_OPTIONS='--max-old-space-size=8192' AGORIC_ESLINT_TYPES=true yarn lint:eslint` in each package.

Unit tests should still pass.

Anything with a change to add explicit `void` will not change any behavior. However, it's possible that there are still hidden bugs resulting from not awaiting or catching these promises. Is it in our best interest to use `void` anywhere then? By doing so, we risk sweeping those bugs under the rug, despite appeasing the lint rule. If we must remove all `void`, then this is a much larger task.

### CODE REVIEW

**Comment by samsiegart** [comment]
CI is failing on `cd packages/wallet/api && yarn lint:types` with errors like:

```
...

../../zoe/tools/types-ambient.js:79:33 - error TS2307: Cannot find module '@agoric/time/src/types' or its corresponding type declarations.

79  * @property {(deadline: import('@agoric/time/src/types').Timestamp,
                                   ~~~~~~~~~~~~~~~~~~~~~~~~

../../zoe/tools/types-ambient.js:147:86 - error TS2307: Cannot find module '@agoric/time/src/types' or its corresponding type declarations.

147  * @returns {{ amountIn: Amount<'nat'>, amountOut: Amount<'nat'>, timestamp?: import('@agoric/time/src/types').Timestamp } | undefined}
                                                                                         ~~~~~~~~~~~~~~~~~~~~~~~~


Found 181 errors in 65 files.
```

I tried running on master and see the same error though.

**Comment by samsiegart** [comment]
I can't run lint in `packages/SwingSet` locally, it seems to get stuck or timeout.

**Comment by samsiegart** [comment]
Some packages, like SwingSet and swingset-runner, seem to get stuck on `NODE_OPTIONS='--max-old-space-size=8192' AGORIC_ESLINT_TYPES=true yarn lint:eslint`. I'm not sure how to proceed @turadg

**Comment by turadg** [comment]
> not sure how to proceed

Maybe due to cycles. Try restricting the linting to subpaths to see what report you can get without hanging.

**Comment by samsiegart** [comment]
>Maybe due to cycles. Try restricting the linting to subpaths to see what report you can get without hanging.

I ran again with `--debug` and it doesn't seem to be hitting any cycles, but for every file it takes a long time and spits out an error like this: https://gist.github.com/samsiegart/cee3e57ed41dee3643d80084247d0666.

Before looking further into this, I'll note to @ivanlei that I've hit the 1-day-of-work timebox for this task. And, I have some doubts (see bottom of PR description) of the benefit of this PR.

**Comment by samsiegart** [comment]
@turadg Ivan says "for stuff that stalls i think just skip them". Given that I've run the command on a per-package basis for each package that doesn't stall, I think it makes sense to merge (after review) these changes at least.

**Comment by turadg** [comment]
> think it makes sense to merge (after review) these changes at least

Fair enough. Please update the description so that it doesn't close the ticket to complete the work. (I was going to say that anyway since we shouldn't close the ticket, an audit, until just before release since the code is still changing.)

**Comment by samsiegart** [comment]
> Please update the description so that it doesn't close the ticket to complete the work.

Done

File: packages/zoe/src/contractFacet/zcfZygote.js [hunk 1]
```diff
@@ -361,7 +361,7 @@ export const makeZCFZygote = async (
       privateArgs = undefined,
     ) => {
       zoeInstanceAdmin = instanceAdminFromZoe;
-      initSeatMgrAndMintFactory();
+      void initSeatMgrAndMintFactory();
```
Comment (turadg) [inline packages/zoe/src/contractFacet/zcfZygote.js]: this should be awaited. if it fails the contract should not restart

File: packages/zoe/src/contractFacet/zcfZygote.js [hunk 2]
```diff
@@ -417,7 +417,7 @@ export const makeZCFZygote = async (
       prepare || Fail`prepare must be defined to upgrade a contract`;
       zoeInstanceAdmin = zcfBaggage.get('zcfInstanceAdmin');
       instanceRecHolder = zcfBaggage.get('instanceRecHolder');
-      initSeatMgrAndMintFactory();
+      void initSeatMgrAndMintFactory();
```
Comment (turadg) [inline packages/zoe/src/contractFacet/zcfZygote.js]: ditto

**Review by turadg — CHANGES_REQUESTED** [review]
I ran into a failure today in zcfZygote because a promise wasn't awaited. I see in here it's marked `void`.

I notice not all promises here are voided but most of them are, so I suspect there are other voids that really should be awaited or error handled. Can you trim down the changes to files you're confident in the way the promises are handled?

File: packages/zoe/src/contractFacet/zcfZygote.js (hunk 1 above)
Comment (samsiegart) [reply packages/zoe/src/contractFacet/zcfZygote.js]: Done

File: packages/zoe/src/contractFacet/zcfZygote.js (hunk 2 above)
Comment (samsiegart) [reply packages/zoe/src/contractFacet/zcfZygote.js]: Done

**Comment by samsiegart** [comment]
Having some trouble with commit history so started a new PR for just the changes I'm confident about https://github.com/Agoric/agoric-sdk/pull/7485

**Pull request closed without merging**